lib/Zonemaster/Backend/DB/MySQL.pm

```diff
@@ -115,6 +115,68 @@ sub create_schema {
         );
     }
 
+    ####################################################################
+    # LOG LEVEL
+    ####################################################################
+    $dbh->do(
+        "CREATE TABLE IF NOT EXISTS log_level (
+            level VARCHAR(15),
+            value INT,
+
+            UNIQUE (level)
+        ) ENGINE=InnoDB
+        "
+    ) or die Zonemaster::Backend::Error::Internal->new( reason => "MySQL error, could not create 'log_level' table", data => $dbh->errstr() );
+
+    my ( $c ) = $dbh->selectrow_array( "SELECT count(*) FROM log_level" );
+    if ( $c == 0 ) {
+        $dbh->do(
+            "INSERT INTO log_level (level, value)
+            VALUES
+                ('DEBUG3'   , -2),
+                ('DEBUG2'   , -1),
+                ('DEBUG'    , 0),
+                ('INFO'     , 1),
+                ('NOTICE'   , 2),
+                ('WARNING'  , 3),
+                ('ERROR'    , 4),
+                ('CRITICAL' , 5)
+            "
+        );
+    }
+
+    ####################################################################
+    # RESULT ENTRIES
+    ####################################################################
+    $dbh->do(
+        "CREATE TABLE IF NOT EXISTS result_entries (
+            hash_id VARCHAR(16) NOT NULL,
+            level VARCHAR(15) NOT NULL,
```

Comment:
Wouldn't it be more efficient to store level directly as an integer so that you could ask for all messages where level is, say, greater than 2? As I understand it now, you have to go via the "log_level" table.

Reply:
I think so. What do others think?